domains/client/domain-executor/src/domain_block_processor.rs

```diff
@@ -700,7 +707,7 @@ where
 
             let fraud_proof = self
                 .fraud_proof_generator
-                .generate_proof(
+                .generate_proof::<PCB>(
```

Comment:
We can use `generate_proof::<PBlock>` here since the source of this generic parameter is essentially:
```rust
pub struct BundleHeader<Number, Hash> {
    /// The block number of primary block at which the bundle was created.
    pub primary_number: Number,
    /// The hash of primary block at which the bundle was created.
    pub primary_hash: Hash,
    ...
}
```

Reply:
I mean the generic type in `FraudProof` is introduced as follows:
```rust
pub enum FraudProof<Number, Hash> {
    ...
    BundleEquivocation(BundleEquivocationProof<Number, Hash>),
}

pub struct BundleEquivocationProof<Number, Hash> {
    ...
    pub first_header: BundleHeader<Number, Hash>,
    pub second_header: BundleHeader<Number, Hash>,
}

pub struct BundleHeader<Number, Hash> {
    pub primary_number: Number,
    pub primary_hash: Hash,
    ...
}
```
In `BundleHeader` there are `primary_number` and `primary_hash`, thus `Block` is always `PBlock`.

Reply:
That's not true, it's the same type as the runtime `Block`, which means it's `PBlock` for the primary chain and `SBlock` for the system domain.

Reply:
I see, I tried using `PBlock` for the `FraudProof` of both the primary chain and system domain and it failed to compile, it seems is due to the runtime API:
```rust
fn submit_fraud_proof_unsigned(fraud_proof: FraudProof<NumberFor<Block>, Block::Hash>);
```
This may be an issue when the core domain tries to construct the `BundleEquivocationProof`, it will have to construct `BundleHeader<NumberFor<SBlock>, SBlock::Hash>` and that is contradicted with the definition of `BundleHeader`:
```rust
pub struct BundleHeader<Number, Hash> {
    pub primary_number: Number,
    pub primary_hash: Hash,
    ...
}
```